network/http/source/main.c

```diff
@@ -5,41 +5,133 @@
 
 #include <3ds.h>
 
-Result http_download(httpcContext *context)//This error handling needs updated with proper text printing once ctrulib itself supports that.
+Result http_download(char *url)
```

Comment:
While the example uses a const char*, this is going to just get copy/pasted into a hundred things that will end up with just a char*. I don't follow the reasoning on why this should be different.